config/jobs/kubernetes/sig-release/release-branch-jobs/1.16.yaml

```diff
@@ -1025,8 +1025,12 @@ presubmits:
         image: gcr.io/k8s-testimages/kubekins-e2e:v20200802-ff98b58-1.16
         name: ""
         resources:
+          limits:
```

Comment:
Do we really need that much, given the numbers for 100-node jobs:
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-scalability/sig-scalability-release-blocking-jobs.yaml#L178-L183

@jkaniuk @jprzychodzen - do we have better estimations on it?

Reply:
I will check after access is granted: https://github.com/kubernetes/k8s.io/pull/1114

Reply:
Well, it seems that the situation is even worse.

For some jobs we were using over 16GiB, with a handful of jobs that crossed 20GiB.

[Used query](https://pantheon.corp.google.com/monitoring/metrics-explorer?organizationId=758905017065&project=k8s-prow&pageState=%7B%22xyChart%22:%7B%22dataSets%22:%5B%7B%22timeSeriesFilter%22:%7B%22filter%22:%22metric.type%3D%5C%22container.googleapis.com%2Fcontainer%2Fmemory%2Fbytes_used%5C%22%5Cn%5Cnresource.type%3D%5C%22gke_container%5C%22%20resource.label.%5C%22container_name%5C%22%3D%5C%22test%5C%22%5Cn%5Cnmetadata.user_labels.%5C%22prow.k8s.io%2Fjob%5C%22%3D%5C%22pull-kubernetes-kubemark-e2e-gce-big%5C%22%22,%22minAlignmentPeriod%22:%2260s%22,%22aggregations%22:%5B%7B%22perSeriesAligner%22:%22ALIGN_MEAN%22,%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22groupByFields%22:%5B%22resource.label.%5C%22pod_id%5C%22%22%5D%7D,%7B%22crossSeriesReducer%22:%22REDUCE_NONE%22%7D%5D%7D,%22targetAxis%22:%22Y1%22,%22plotType%22:%22LINE%22%7D%5D,%22options%22:%7B%22mode%22:%22COLOR%22%7D,%22constantLines%22:%5B%5D,%22timeshiftDuration%22:%220s%22,%22y1Axis%22:%7B%22label%22:%22y1Axis%22,%22scale%22:%22LINEAR%22%7D%7D,%22isAutoRefresh%22:true,%22timeSelection%22:%7B%22timeRange%22:%221m%22%7D,%22xZoomDomain%22:%7B%22start%22:%222020-07-21T16:09:10.184Z%22,%22end%22:%222020-07-23T00:17:44.470Z%22%7D%7D)

Screenshot
![PCOFXOSPGk7](https://user-images.githubusercontent.com/29059230/89537321-77811c00-d7f9-11ea-95cc-d99b59aee355.png)

Reply:
@jprzychodzen I think we can set 24Gi memory from the observation.

Reply:
Hi folks, should I increase memory requests/limits in my PR? Thx.

Reply:
Actually, I think that 13GB is right on mark:
![image](https://user-images.githubusercontent.com/5487158/89543390-25dc8f80-d801-11ea-8fc9-203ac82ec3fd.png)
[Used query](https://pantheon.corp.google.com/monitoring/metrics-explorer?organizationId=758905017065&project=k8s-prow&pageState=%7B%22xyChart%22:%7B%22dataSets%22:%5B%7B%22timeSeriesFilter%22:%7B%22filter%22:%22metric.type%3D%5C%22container.googleapis.com%2Fcontainer%2Fmemory%2Fbytes_used%5C%22%20resource.type%3D%5C%22gke_container%5C%22%20metadata.user_labels.%5C%22prow.k8s.io%2Fjob%5C%22%3Dmonitoring.regex.full_match(%5C%22pull-kubernetes-kubemark.*%5C%22)%22,%22minAlignmentPeriod%22:%2260s%22,%22unitOverride%22:%22By%22,%22aggregations%22:%5B%7B%22perSeriesAligner%22:%22ALIGN_MEAN%22,%22crossSeriesReducer%22:%22REDUCE_MAX%22,%22groupByFields%22:%5B%22metadata.user_labels.%5C%22prow.k8s.io%2Fjob%5C%22%22,%22resource.label.%5C%22cluster_name%5C%22%22,%22resource.label.%5C%22pod_id%5C%22%22%5D%7D,%7B%22crossSeriesReducer%22:%22REDUCE_NONE%22%7D%5D%7D,%22targetAxis%22:%22Y1%22,%22plotType%22:%22LINE%22%7D%5D,%22options%22:%7B%22mode%22:%22COLOR%22%7D,%22constantLines%22:%5B%5D,%22timeshiftDuration%22:%220s%22,%22y1Axis%22:%7B%22label%22:%22y1Axis%22,%22scale%22:%22LINEAR%22%7D%7D,%22isAutoRefresh%22:true,%22timeSelection%22:%7B%22timeRange%22:%221m%22%7D%7D)

Reply:
@jprzychodzen - Could you check again?

Reply:
OK, taking that back, its around 20GB.
Evalle - please increase those limits.

Reply:
It seams that you are using `max`, instead of `sums` aggregator - we should set the requirements for sum of evictable and non-evictable memory.

Reply:
@pontiyaraja It seems that 24GBi should be ok. Let's increase it in this PR.

Reply:
This is ridiculous - it's more than the 5k-node jobs is using:
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-scalability/sig-scalability-release-blocking-jobs.yaml#L106

@jprzychodzen @jkaniuk - I can't believe we need more for 500-node kubmeark than for 5k reaal node tests (objects in kubemark at not larger than in 5k, there are less of them, the test is shorter, throughput is the same).
How is it possible that we need that much for it?

@mm4tt ^^

Reply:
Yeah, it's strange...

Jacek and Kuba, are you sure that your estimation method is correct? What do you get if you do the analysis for the gce-5k-performance job?

Reply:
**tl;dr We can set around 8GB of ram and about 6 cpus and use `flush-mem-after-build` flag for kubetest**

I've done some digging around, and it seems that memory consumption is mostly related to bazel build. Even after process is killed, some cache allocations leave a trace unless `sync` is used (already implemented in kubetest).

It's hard to say what would be best value of memory limit for build phase, however after that process is no longer memory hungry, so I think that we can go with much lower value - 8Gi.

Regarding CPU - because there is a build happening, we should have at least 6 vCPUs. This should be sufficient for our needs right now. Again, post-build phase is not CPU sensitive and could go to much lower value, like singe CPU.

Reply:
@jprzychodzen @jkaniuk #18530 here is the complete list of jobs to be analysed, Please check those jobs and update the resources in that issue.

Reply:
Ok should I change my commit to 6cpus, 8Gb of ram and add `- --flush-mem-after-build=true` to the jobs?